torch/csrc/jit/codegen/cuda/runtime/grid_reduction.cu

```diff
@@ -294,6 +295,72 @@ __device__ void gridReduce(
   }
 }
 
+// This is just a wrapper of the above grid reduction routine to
+// measure the elapsed cycles. The measurement must be done just by
+// one thread, and in this case it should be done by one of the
+// threads in the last thread block.
+#ifdef PYTORCH_NVFUSER_PROFILE_KERNEL
+template <
+    bool X_BLOCK,
+    bool Y_BLOCK,
+    bool Z_BLOCK,
+    bool X_THREAD,
+    bool Y_THREAD,
+    bool Z_THREAD,
+    bool PERSISTENT_REDUCTION,
+    typename T,
+    typename Func>
+__device__ void gridReduce(
```

Comment:
Is there a cleaner mechanism in C++ to forward all arguments except the last two while also using template forwarding? I'm guessing maybe no, but might be cleaner.

Reply:
I did try `std::forward`, but it's not available with nvrtc (haven't tried with nvcc). We could probably develop something similar to `std::forward` by ourselves, but I'm not really confident especially in the device side with nvrtc.

Reply:
Okay, doesn't seem worth it.